staging/src/k8s.io/kubectl/pkg/cmd/attach/attach.go

```diff
@@ -55,13 +55,98 @@ var (
 		# Get output from the first pod of a ReplicaSet named nginx
 		kubectl attach rs/nginx
 		`))
+
+	defaultAttachResourceBuilderFlags = genericclioptions.NewResourceBuilderFlags().
+						WithLabelSelector("").
+						WithFieldSelector("").
+						WithAll(false).
+						WithAllNamespaces(false).
+						WithLocal(false).
+						WithLatest()
 )
 
 const (
 	defaultPodAttachTimeout = 60 * time.Second
 	defaultPodLogsTimeout   = 20 * time.Second
 )
 
+// AttachFlags directly reflect the information that CLI is gathering via
+// flags. They will be converted to Options, which reflect the runtime
+// requirements for the command.
+type AttachFlags struct {
+	builder              func() *resource.Builder
+	resourceBuilderFlags *genericclioptions.ResourceBuilderFlags
+	restClientGetter     genericclioptions.RESTClientGetter
+	exec.StreamOptions
+}
+
+// NewAttachFlags returns a default AttachFlags.
+func NewAttachFlags(
+	f cmdutil.Factory,
+	streams genericclioptions.IOStreams) *AttachFlags {
+
+	return &AttachFlags{
+		builder:              f.NewBuilder,
+		resourceBuilderFlags: defaultAttachResourceBuilderFlags,
+		restClientGetter:     f,
+		StreamOptions: exec.StreamOptions{
+			IOStreams: streams,
+		},
+	}
+}
+
+// ToOptions convert the attach flags to options.
+func (flags *AttachFlags) ToOptions(
```

Comment:
This `ToOptions()` method replaces the old `NewAttachOptions()`, `Validate()` and `Complete()` functions.